devices/linkam_t95/device.py

```diff
@@ -0,0 +1,183 @@
+            (('started', 'hold'), lambda: self._context.temperature == self._context.temperature_limit),
+            (('started', 'cool'), lambda: self._context.temperature > self._context.temperature_limit),
+
+            (('heat', 'hold'), lambda: self._context.temperature == self._context.temperature_limit or self._context.hold_commanded),
+            (('heat', 'cool'), lambda: self._context.temperature > self._context.temperature_limit),
+            (('heat', 'stopped'), lambda: self._context.stop_commanded),
+
+            (('hold', 'heat'), lambda: self._context.temperature < self._context.temperature_limit and not self._context.hold_commanded),
+            (('hold', 'cool'), lambda: self._context.temperature > self._context.temperature_limit and not self._context.hold_commanded),
+            (('hold', 'stopped'), lambda: self._context.stop_commanded),
+
+            (('cool', 'heat'), lambda: self._context.temperature < self._context.temperature_limit),
+            (('cool', 'hold'), lambda: self._context.temperature == self._context.temperature_limit or self._context.hold_commanded),
+            (('cool', 'stopped'), lambda: self._context.stop_commanded),
+        ])
+
+        if override_transitions is not None:
+            dict_strict_update(transition_handlers, override_transitions)
+
+        self._csm = StateMachine({
+            'initial': 'init',
+            'states': state_handlers,
+            'transitions': transition_handlers,
+        }, context=self._context)
+
+        self.addProcessor(self._csm)
+
+    def getStatus(self):
+        self._context.serial_command_mode = True
+
+        Tarray = [0x80] * 10
+
+        # Status byte (SB1)
+        Tarray[0] = {
+            'stopped': 0x01,
+            'heat': 0x10,
+            'cool': 0x20,
+            'hold': 0x30,
+        }.get(self._csm.state, 0x01)
+        if Tarray[0] == 0x30 and self._context.hold_commanded:
+            Tarray[0] = 0x40 if self._context.temperature == self._context.temperature_limit else 0x50
+
+        # Error status byte (EB1)
+        if self._context.pump_overspeed:
+            Tarray[1] |= 0x01
+        # TODO: Add support for other error conditions?
+
+        # Pump status byte (PB1)
+        Tarray[2] = 0x80 + self._context.pump_speed
+
+        # Temperature
+        Tarray[6:10] = [ord(x) for x in "%04x" % (int(self._context.temperature * 10) & 0xFFFF)]
+
+        print self._csm.state
+        print str(Tarray)
+
+        return ''.join(chr(c) for c in Tarray)
+
+    def setRate(self, param):
+        # TODO: Is not having leading zeroes / 4 digits an error?
+        # TODO: What is the upper limit in the real device?
+        rate = int(param)
+        if 1 <= rate <= 9999:
+            self._context.temperature_rate = rate / 100.0
+        print "New rate: %.2f C/min" % (self._context.temperature_rate,)
+        return ""
+
+    def setLimit(self, param):
+        # TODO: Is not having leading zeroes / 4 digits an error?
```

Comment:
Well, I'm not so certain one way or the other...

The -1960 to 15000 values are from the documented limits on the temperature readout. They're not necessarily correct for the input. The documentation for the input doesn't specify limits or whether there should be leading zeroes. It only provides an example setting the rate to 2000, and the limit to 1250 and -1250.

The .proto file used by IBEX, on the other hand, uses sprintf style formatting that implies four digits (not counting the `-` sign) and leading zeroes. No indication for upper or lower limits there either.